distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == "application/vnd.docker.plugin.v0+json" {
```

Comment:
Yes, `"application/vnd.docker.plugin.v0+json"` is indeed the right mediaType that image pull should check and error out.